web/packages/teleport/src/AuthConnectors/AuthConnectorEditor/GitHubConnectorEditor.tsx

```diff
@@ -0,0 +1,92 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect, useState } from 'react';
+import { useHistory, useParams } from 'react-router';
+
+import useAttempt from 'shared/hooks/useAttemptNext';
+
+import cfg from 'teleport/config';
+import useTeleport from 'teleport/useTeleport';
+
+import templates from '../templates';
+import { AuthConnectorEditorContent } from './AuthConnectorEditorContent';
+
+/**
+ * GitHubConnectorEditor is the edit/create page for a GitHub Auth Connector.
+ */
+export function GitHubConnectorEditor({ isNew = false }) {
+  const { connectorName } = useParams<{
+    connectorName: string;
+  }>();
+  const ctx = useTeleport();
+  const history = useHistory();
+
+  const [content, setContent] = useState(templates['github']);
+  const [initialContent, setInitialContent] = useState(templates['github']);
+
+  const fetchAttempt = useAttempt(isNew ? 'success' : 'processing');
```

Comment:
Aren't we now favoring `useAsync` from `shared/hooks/useAsync` over `useAttemptNext`?

`useAsync` returns attempt and run which you can then call below in useEffect.

Same comment for `saveAttempt` below and in other files.